src-tests/suite.lisp

```diff
@@ -74,11 +74,34 @@
   (let* ((original (make-instance 'rpcq::|RPCRequest|
                                   :|method| "test-method"
                                   :|params| (make-hash-table)
-                                  :|id| "test-id"))
+                                  :|id| "test-with-empty-params"))
          (cloned (rpcq::deserialize (rpcq::serialize original))))
     (is (typep cloned 'rpcq::|RPCRequest|))
     (is (string= (rpcq::|RPCRequest-id| original)     (rpcq::|RPCRequest-id| cloned)))
-    (is (string= (rpcq::|RPCRequest-method| original) (rpcq::|RPCRequest-method| cloned))))
+    (is (string= (rpcq::|RPCRequest-method| original) (rpcq::|RPCRequest-method| cloned)))
+    (is (= 0
+           (hash-table-count (rpcq::|RPCRequest-params| original))
+           (hash-table-count (rpcq::|RPCRequest-params| cloned)))))
+
+  (let* ((original (make-instance 'rpcq::|RPCRequest|
+                                  :|method| "test-method"
+                                  :|params| (rpcq::prepare-rpc-call-args
+                                             '("a1" a2 42 :kw1 "k1" :kw-2 "k2"))
+                                  :|id| "test-with-non-empty-params"))
+         (cloned (rpcq::deserialize (rpcq::serialize original)))
+         (original-params (rpcq::|RPCRequest-params| original))
+         (cloned-params (rpcq::|RPCRequest-params| cloned)))
+
+    ;; We can't simply test (EQUALP ORIGINAL-PARAMS CLONED-PARAMS) here because of the 'A2 symbol in
+    ;; *args, which gets deserialized as the STRING "A2".
```

Comment:
Should I also change `PREPARE-RPC-CALL-ARGS` to call `SANTIZE-NAME` on non-keyword symbol args or leave this as-is? That would result in the argument `'foo-bar` getting serialized/deserialized to `"foo_bar"` rather than `"FOO-BAR"`.

Reply:
I have a preference for `deserialize . serialize` to be the identity. What is the argument for sanitizing these? If we were producing python code I would be on-board, but we're just passing data right?

Reply:
Only for symmetry with how keyword symbols are handled, plus maybe simplifying this portion of the test to `(EQUALP ORIGINAL-PARAMS CLONED-PARAMS)`. But this is a minor concern.

I also like serialize -> deserialize to be identity, but not sure how that would work with the need to be compatible with Python in this case. Currently, there is no `%serialize` method specialized for `symbol`s.  I think we just punt to `messagepack:encode`, which encodes it as a string.

```lisp
(defun encode-symbol (data stream)
  ;; TODO: bind *package* to :keywords, so that the package is included as well?
  (encode-string (symbol-name data) stream))
```

Reply:
Forgot to mention: maybe I'm just abusing rpcq here, and the answer is "don't do that", meaning don't attempt to pass a symbol in the param list other than keywords.

Reply:
I think abuse is the right diagnosis: Python doesn't have an analogue of Lisp's symbol type, so we don't expect to be able to faithfully de/ser symbols.

I'd like to call `sanitize-name` on things that turn into "code": function names, kwarg names, that sort of thing.